### ISSUE TITLE
Illegal instruction in test code

### ISSUE THREAD

**Issue by KANG-BIOINFO** [frankligy/scTriangulate#3]
It seems there's some weird bug in test.py code. Could you take a look and see why it happens? I ran it on my Mac:
```
P14-5847:sctriangulate_test kang$ /Users/kang/miniconda3/envs/aipy/bin/python3.7 -W ignore mini_test.py 
===================
import modules test: ✓ 
====================
Preprocessing...
Simulating doublets...
Embedding transcriptomes using PCA...
Calculating doublet scores...
Illegal instruction: 4
```

**Comment by frankligy** [comment]
It seems to be a conda version issue, mine is conda 4.10.1 and my mac version is 11.4
See (https://github.com/conda/conda/issues/9678)

I carefully compare our dependencies difference 
[dependency_mac.txt](https://github.com/frankligy/scTriangulate/files/7318076/dependency_mac.txt)

**Comment by frankligy** [comment]
Hello @KANG-BIOINFO just want to give you an update here, as I realize that the another potential cause is due to `scrublet` package, look at this issue (https://github.com/swolock/scrublet/issues/28). I experienced that same issue when using `annoy==1.7.0`, downgrading to 1.6.3 indeed help me to pass through this hurdle.

With that being said, I used annoy==1.7.0 in my python=3.6 environment and it works well, so it is still mysterious to me at this time point that I am going to keep looking into it.
